File: src/main/java/br/com/oficina/oficina/model/Atendimento.java
package br.com.oficina.oficina.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "atendimentos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Atendimento {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(columnDefinition = "TEXT")
    private String descricaoServico;

    @Column(name = "data_entrada")
    private LocalDateTime dataEntrada;

    @Column(name = "data_conclusao")
    private LocalDateTime dataConclusao;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusAtendimento status;

    @Column(name = "data_cadastro")
    private LocalDateTime dataCadastro;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", nullable = false)
    @JsonIgnore
    private Cliente cliente;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "veiculo_id", nullable = false)
    @JsonIgnore
    private Veiculo veiculo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "funcionario_id", nullable = false)
    @JsonIgnore
    private Funcionario funcionario;

    @PrePersist
    protected void onCreate() {
        dataCadastro = LocalDateTime.now();
        if (dataEntrada == null) {
            dataEntrada = LocalDateTime.now();
        }
        if (status == null) {
            status = StatusAtendimento.AGUARDANDO;
        }
    }
}

File: crm-oficina/src/types/atendimento.ts
export const StatusAtendimento = {
  AGUARDANDO: 'AGUARDANDO',
  ANDAMENTO: 'ANDAMENTO',
  CONCLUIDO: 'CONCLUIDO',
  CANCELADO: 'CANCELADO',
} as const

export type StatusAtendimento = (typeof StatusAtendimento)[keyof typeof StatusAtendimento]

export const STATUS_LABELS: Record<StatusAtendimento, string> = {
  [StatusAtendimento.AGUARDANDO]: 'Aguardando',
  [StatusAtendimento.ANDAMENTO]: 'Em Andamento',
  [StatusAtendimento.CONCLUIDO]: 'Concluído',
  [StatusAtendimento.CANCELADO]: 'Cancelado',
}

export const STATUS_ORDER: StatusAtendimento[] = [
  StatusAtendimento.AGUARDANDO,
  StatusAtendimento.ANDAMENTO,
  StatusAtendimento.CONCLUIDO,
  StatusAtendimento.CANCELADO,
]

export interface AtendimentoRequest {
  descricaoServico: string
  statusAtendimento: string
  clienteId: string
  veiculoPlaca: string
  funcionarioId: string
}

export interface AtendimentoResponse {
  id: string
  descricaoServico: string
  dataEntrada: string
  dataConclusao: string | null
  status: StatusAtendimento
  dataCadastro: string
  cliente: string
  veiculo: string
  funcionario: string
}
